src/pytest_cov/plugin.py

```diff
@@ -41,11 +41,52 @@ def validate_report(arg):
     return values
 
 
-def validate_fail_under(num_str):
-    try:
-        return int(num_str)
-    except ValueError:
-        return float(num_str)
+class FailUnder(object):
+    @staticmethod
+    def __number(num_str):
+        if num_str.endswith('%'):
+            num_str = num_str[:-1]
+        try:
+            return int(num_str)
+        except ValueError:
+            return float(num_str)
+
+    @classmethod
+    def parse(cls, unparsed):
+        items = iter(unparsed.split(":"))
+        limit = cls.__number(next(items, None))
```

Comment:
We could use the validate_fail_under from the other merged PR here.

Reply:
That's gone now